shell/platform/fuchsia/flutter/vsync_waiter_unittest.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "flutter/fml/task_runner.h"
+#include "flutter/shell/common/thread_host.h"
+#include "fml/make_copyable.h"
+#include "fml/message_loop.h"
+#include "fml/synchronization/waitable_event.h"
+#include "fml/time/time_delta.h"
+#include "fml/time/time_point.h"
+#include "vsync_waiter.h"
+
+namespace flutter_runner {
+
+TEST(VSyncWaiterFuchsia, FrameScheduledForStartTime) {
+  using flutter::ThreadHost;
+  std::string prefix = "vsync_waiter_test";
+
+  fml::MessageLoop::EnsureInitializedForCurrentThread();
+  auto platform_task_runner = fml::MessageLoop::GetCurrent().GetTaskRunner();
```

Comment:
This test is failing because the the unit-tests don't always have a default task runner setup. The way to fix this would be to create a thread (you are already using a thread host so add `flutter::ThreadHost::Type::PLATFORM` to it and and use its task runner). Please don't `FML_USED_ON_EMBEDDER` as that is may not work in all unit-test.

Reply:
Oh, ok. Disregard the latest push. Amended FML_USED_ON_EMBEDDER before seeing your comment.

Reply:
Yeah, that define is used by embedder platforms that depend on a default message loop already setup. The engine internals don't use that by default and we added the deprecation explicitly to trip accidentally calling the accessor instead of holding references to a task runner.